datalake/src/main/java/com/sequenceiq/datalake/flow/create/handler/RdsWaitHandler.java

```diff
@@ -0,0 +1,110 @@
+import com.sequenceiq.environment.api.v1.environment.model.response.SecurityAccessResponse;
+import com.sequenceiq.redbeams.api.endpoint.v4.databaseserver.responses.DatabaseServerStatusV4Response;
+
+@Component
+public class RdsWaitHandler extends ExceptionCatcherEventHandler<RdsWaitRequest> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RdsWaitHandler.class);
+
+    @Inject
+    private DatabaseService databaseService;
+
+    @Inject
+    private SdxClusterRepository sdxClusterRepository;
+
+    @Override
+    protected Selectable defaultFailureEvent(Long resourceId, Exception e) {
+        return new SdxCreateFailedEvent(resourceId, null, e);
+    }
+
+    @Override
+    public String selector() {
+        return "RdsWaitRequest";
+    }
+
+    @Override
+    protected void doAccept(HandlerEvent event) {
+        RdsWaitRequest rdsWaitRequest = event.getData();
+        Long sdxId = rdsWaitRequest.getResourceId();
+        Optional<SdxCluster> sdxCluster = sdxClusterRepository.findById(sdxId);
+        String userId = rdsWaitRequest.getUserId();
+        DetailedEnvironmentResponse env = rdsWaitRequest.getDetailedEnvironmentResponse();
+        Selectable response;
+        try {
+            if (sdxCluster.map(SdxCluster::getCreateDatabase).orElse(Boolean.FALSE)) {
+                validForDatabaseCreation(sdxId, env);
+                LOGGER.debug("start polling database for sdx: {}", sdxId);
+                DatabaseServerStatusV4Response db = databaseService.create(sdxId, sdxCluster, env);
+                response = new RdsWaitSuccessEvent(sdxId, userId, env, db);
+            } else {
+                LOGGER.debug("skipping creation of database for sdx: {}", sdxId);
+                response = new RdsWaitSuccessEvent(sdxId, userId, env, null);
+            }
+        } catch (UserBreakException userBreakException) {
+            LOGGER.info("Database polling exited before timeout. Cause: ", userBreakException);
+            response = new SdxCreateFailedEvent(sdxId, userId, userBreakException);
+        } catch (PollerStoppedException pollerStoppedException) {
+            LOGGER.info("Database poller stopped for sdx: {}", sdxId, pollerStoppedException);
+            response = new SdxCreateFailedEvent(sdxId, userId, pollerStoppedException);
+        } catch (PollerException exception) {
+            LOGGER.info("Database polling failed for sdx: {}", sdxId, exception);
+            response = new SdxCreateFailedEvent(sdxId, userId, exception);
+        } catch (Exception anotherException) {
+            LOGGER.error("Something wrong happened in sdx database creation wait phase", anotherException);
+            response = new SdxCreateFailedEvent(sdxId, userId, anotherException);
+        }
+        sendEvent(response, event);
+    }
+
+    private void validForDatabaseCreation(Long sdxId, DetailedEnvironmentResponse env) {
```

Comment:
please move validation before flow start and validate it when request received

Reply:
For the validation we need the environment response and that is only available in the flow

Reply:
We have environment response here:
https://github.com/hortonworks/cloudbreak/blob/0a4b4d41adce1792d354a8d72d23aec8132027be/datalake/src/main/java/com/sequenceiq/datalake/service/sdx/SdxService.java#L129